### DISCUSSION TITLE
References missing in REST response

### DISCUSSION THREAD

**Discussion by eskhisov** [Apicurio/apicurio-registry#4876] [Q&A]
I have the following piece of code to add references to the existing AVRO schema:
```
List<ArtifactReference> references = this.client.getArtifactReferencesByGlobalId(globalId);
references.add(new ArtifactReference(referenceGroupId, referenceArtifactId, referenceVersion,
                                 this.getArtifactName(referenceArtifactId)));
this.client.updateArtifact(this.getGroupIdById(artifactId), artifactId,
                newVersion, artifact.getName(),
                artifact.getDescription(), this.client.getLatestArtifact(groupId, artifactId), references);

```
When I check UI after running my code, the references are there:

![image (9)](https://github.com/Apicurio/apicurio-registry/assets/4778589/cec9096f-bc97-4ae9-bd90-1d7b8e09e027)
However, when I retrieve the same data with API call, references are empty:
```
{
  "count": 3,
  "versions": [
    {
      "name": "AgentStateSegment",
      "description": "",
      "createdOn": "2024-03-14T18:20:02.000+00:00",
      "createdBy": "",
      "type": "AVRO",
      "labels": [
        "ACD"
      ],
      "state": "ENABLED",
      "globalId": 26,
      "version": "1",
      "properties": {
        "date": "June 24"
      },
      "contentId": 19,
      "references": []
    },
    {
      "name": "AgentStateSegment",
      "description": "",
      "createdOn": "2024-07-09T19:30:56.000+00:00",
      "createdBy": "",
      "type": "AVRO",
      "labels": [
        "ACD"
      ],
      "state": "ENABLED",
      "globalId": 47,
      "version": "2",
      "properties": {
        "date": "June 24"
      },
      "contentId": 36,
      "references": []
    },
    {
      "name": "AgentStateSegment",
      "description": "",
      "createdOn": "2024-07-09T20:29:04.000+00:00",
      "createdBy": "",
      "type": "AVRO",
      "labels": [
        "ACD"
      ],
      "state": "ENABLED",
      "globalId": 48,
      "version": "3",
      "properties": {
        "date": "June 24"
      },
      "contentId": 37,
      "references": []
    }
  ]
}
```

Is this a bug, or am I doing something wrong?

**Comment by EricWittmann** [comment]
That might be a bug in the Get Versions endpoint. What happens if you get a single version?

**Comment by cesareomacias** [comment]
Hi @eskhisov 

I don't know which version of Apicurio Registry are you using, neither the endpoint that you're requesting, but I'd say, according to the documentation, the proper endpoint should be:

`/groups/{groupId}/artifacts/{artifactId}/versions/{version}/references`
_"Retrieves all references for a single version of an artifact. Both the artifactId and the unique version number must be provided. Using the refType query parameter, it is possible to retrieve an array of either the inbound or outbound references."_

Regards!

**Comment by carlesarnal** [comment]
I have created an issue from this. It seems indeed like a bug. The references in the UI are there because it's using the endpoint @cesareomacias mentioned.

**Reply by EricWittmann** [reply]
It may be a bug - although should we return references in search results?  Maybe we should not.
